modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/healthcheck/declaration/PeerDeclarationHealthCheck.scala

```diff
@@ -66,6 +66,20 @@ object PeerDeclarationHealthCheck {
             _ <- roundKeys.traverse(startOwnRound)
           } yield ()
 
+        def onOutcome(peers: ConsensusRounds.Outcome[F, Key[K], Health, Status[K]]): F[Unit] =
+          peers.toList.traverse {
+            case (key, t) =>
+              t match {
+                case (decision, _) if decision.isInstanceOf[PositiveOutcome] =>
+                  logger.info(s"Outcome for peer ${key.id}: positive - no action required")
+                case (decision, _) if decision.isInstanceOf[NegativeOutcome] =>
+                  logger.info(s"Outcome for peer ${key.id}: negative - removing peer") >>
+                    clusterStorage.removePeer(key.id)
+                case (decision, _) =>
```

Comment:
It's not very important, but it would be more elegant with a sealed trait - you could have an exhaustive match with just 2 cases. It would require another type parameter to the `HealthCheckConsensus` class, but maybe it's worth it.